nirc_ehr/src/org/labkey/nirc_ehr/NIRCOrchardFileGenerator.java

```diff
@@ -0,0 +1,110 @@
+package org.labkey.nirc_ehr;
+
+import org.labkey.api.data.Container;
+import org.labkey.api.data.DbScope;
+import org.labkey.api.data.SimpleFilter;
+import org.labkey.api.data.TableInfo;
+import org.labkey.api.data.TableSelector;
+import org.labkey.api.ehr.EHRService;
+import org.labkey.api.module.Module;
+import org.labkey.api.module.ModuleLoader;
+import org.labkey.api.module.ModuleProperty;
+import org.labkey.api.query.FieldKey;
+import org.labkey.api.query.QueryService;
+import org.labkey.api.query.UserSchema;
+import org.labkey.api.security.User;
+import org.labkey.api.study.StudyService;
+import org.labkey.api.util.JobRunner;
+import org.labkey.api.util.PageFlowUtil;
+import org.labkey.api.writer.PrintWriters;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+
+public class NIRCOrchardFileGenerator
+{
+    private Container _container = null;
+    private User _user = null;
+
+    public static final String NIRCOrchardFileLocation = "NIRCOrchardFileLocation";
+    private static final String orchardFileName = "orchardFile.txt";
+
+    public NIRCOrchardFileGenerator(Container c, User u)
+    {
+        _container = c;
+        _user = u;
+    }
+
+    public void generateOrchardFile(Container c, String animalId)
+    {
+        String orchardFileLocation = getOrchardFileLocation(c);
+        // Generate the Orchard file
+        if (orchardFileLocation == null)
+        {
+            return;
+        }
+        try (DbScope.Transaction transaction = StudyService.get().getDatasetSchema().getScope().ensureTransaction())
+        {
+            // Add post commit task to run provider update in another thread once this transaction is complete.
```

Comment:
This isn't doing a provider update. Can you refresh the comment? I'm curious why this is a commit task and why it's being done in a separate thread.

Reply:
Also, if a single transaction processes more than one animal ID, won't it write out one file per ID, each time stomping over the file it wrote immediately before?

Reply:
@labkey-jeckels This is going to be kicked off in various triggers and some of the data in the transaction will be needed in the file. So the cleanest way to ensure some yet unspecified data was committed to the db was to do this post commit. And a separate thread just to not hold up the main UI thread waiting to generate the file.

Writing the file and its contents itself will be left up to the external dev here, but @ankurjuneja can we add the animal Id or something to the file name to just show how an update with multiple animals might work. And update the comment.

Reply:
And thinking about it, we should probably just get a list of ids from the onUpsert triggers and do the final call to this trigger helper in onComplete with a set of unique Ids.